src/main/scala/edu/ie3/simona/model/participant/PvModel.scala

```diff
@@ -51,45 +51,35 @@ final case class PvModel private (
     with ApparentPowerParticipant[PvRelevantData, ConstantState.type] {
 
   /** Override sMax as the power output of a pv unit could become easily up to
-    * 10% higher than the sRated value found in the technical sheets
+    * 10% higher than the sRated value found in the technical sheets.
     */
   override val sMax: ApparentPower = sRated * 1.1
 
-  /** Permissible maximum active power feed in (therefore negative) */
+  /** Permissible maximum active power feed in (therefore negative). */
   protected val pMax: Power = sMax.toActivePower(cosPhiRated) * -1d
 
-  /** Reference yield at standard testing conditions (STC) */
+  /** Reference yield at standard testing conditions (STC). */
   private val yieldSTC = WattsPerSquareMeter(1000d)
 
   private val activationThreshold =
     sRated.toActivePower(cosPhiRated) * 0.001 * -1d
 
-  /** Calculate the active power behaviour of the model
+  /** Calculate the active power behaviour of the model.
     *
     * @param data
-    *   Further needed, secondary data
+    *   Further needed, secondary data.
     * @return
-    *   Active power
+    *   THe Active power.
     */
   override protected def calculateActivePower(
       modelState: ConstantState.type,
       data: PvRelevantData,
   ): Power = {
-    // === Weather Base Data  === //
-    /* The pv model calculates the power in-feed based on the solar irradiance that is received over a specific
-     * time frame (which actually is the solar irradiation). Hence, a multiplication with the time frame within
-     * this irradiance is received is required. */
-    val duration: Time = Seconds(data.weatherDataFrameLength)
-
-    // eBeamH and eDifH needs to be extract to their double values in some places
-    // hence a conversion to watt-hour per square meter is required, to avoid
-    // invalid double value extraction!
-    val eBeamH =
-      data.dirIrradiance * duration
-    val eDifH =
-      data.diffIrradiance * duration
-
-    // === Beam Radiation Parameters  === //
+    // Irradiance on a horizontal surface
+    val gBeamH = data.dirIrradiance
+    val gDifH = data.diffIrradiance
```

Comment:
Does this work if we have weather data with different step sizes?

Reply:
That's exactly the thing we're solving here. With irradiance, we don't care about the step size anymore.